lib/core/ingestion/es_sink.rb

```diff
@@ -11,49 +11,108 @@
 require 'utility/es_client'
 require 'utility/logger'
 require 'elasticsearch/api'
-
+#
+# This class is responsible for sending the data to the data storage.
+# While we don't actually allow to output our data anywhere except
+# Elasticsearch, we still want to be able to do so sometime in future.
+#
+# This class should stay simple and any change to the class should be careful
+# with the thought of introducing other sinks in future.
 module Core
   module Ingestion
     class EsSink
-      def initialize(index_name, request_pipeline, bulk_queue = Utility::BulkQueue.new)
+      def initialize(index_name, request_pipeline, bulk_queue = Utility::BulkQueue.new, max_allowed_document_size = 5 * 1024 * 1024)
```

Comment:
I suggest adding the `5*1024*1024` to the `constants.rb` file

Reply:
It'll actually be addressed in a follow-up PR to introduce this as a configuration provided by YAML file, I'll do it there!